pkg/compact/compact.go

```diff
@@ -470,6 +483,200 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// metrics related to planning and compaction progress
+type ProgressMetrics struct {
+	NumberOfCompactionRuns   *prometheus.GaugeVec
+	NumberOfCompactionBlocks *prometheus.GaugeVec
+}
+
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	ProgressCalculate(ctx context.Context, groups []*Group) error
+}
+
+type DefaultPlanSim struct {
```

Comment:
We don't need to call it `default`. What about `CompactionSimulator` or `CompactionCalculator`?

Reply:
Yes, that sounds better. I had used `default` looking at some other functions/structs in the codebase.